Sources/SwiftSyntax/generated/SyntaxBaseNodes.swift

```diff
@@ -90,49 +84,55 @@ extension DeclSyntaxProtocol {
+  /// than ``DeclSyntaxProtocol``.
   ///
-  /// - Returns: `false` since the node can not be cast to the node type from different base node protocol hierarchy than ``DeclSyntaxProtocol``.
+  /// - Returns: `true` if the node can be cast, `false` otherwise.
   ///
-  /// - Note: This method overloads the general `is` method and is marked as deprecated to produce a warning,
-  ///         informing the user that the cast will always fail.
-  @available(*, deprecated, message: "This cast will always fail")
+  /// - Note: This generally does not make sense to call - if in a generic ``SyntaxProtocol`` context, use
+  /// ``DeclSyntaxProtocol`` instead. Otherwise this must be a node outside of ``DeclSyntaxProtocol``,
+  /// in which case this call will always fail.
+  @available(*, deprecated, message: "Type should be part of the 'DeclSyntaxProtocol' hierarchy")
   public func `is`<S: SyntaxProtocol>(_ syntaxType: S.Type) -> Bool {
-    return false
+    return self.as(syntaxType) != nil
   }
   
 
-  /// Attempts to cast the current syntax node to a given node type from the different base node protocol hierarchy than ``DeclSyntaxProtocol``.
+  /// Attempts to cast the current syntax node to a given node type from the a base node protocol hierarchy other than
+  /// ``DeclSyntaxProtocol``. This is generally does not make sense to call - if in a generic ``SyntaxProtocol``
+  /// context, use ``DeclSyntaxProtocol`` instead. Otherwise this must be a base node other than
+  /// ``DeclSyntaxProtocol``, in which case this cast will always return `nil`.
   ///
-  /// - Returns: `nil` since the node can not be cast to the node type from different base node protocol hierarchy than ``DeclSyntaxProtocol``.
+  /// - Returns: An instance of the specialized type, or `nil` if the cast fails.
   ///
-  /// - Note: This method overloads the general `as` method and is marked as deprecated to produce a warning,
-  ///         informing the user that the cast will always fail.
-  @available(*, deprecated, message: "This cast will always fail")
+  /// - Note: This generally does not make sense to call - if in a generic ``SyntaxProtocol`` context, use
+  /// ``DeclSyntaxProtocol`` instead. Otherwise this must be a node outside of ``DeclSyntaxProtocol``,
+  /// in which case this cast will always return `nil`.
+  @available(*, deprecated, message: "Type should be part of the 'DeclSyntaxProtocol' hierarchy")
   public func `as`<S: SyntaxProtocol>(_ syntaxType: S.Type) -> S? {
-    return nil
+    return S.init(self)
   }
   
 
-  /// Force-casts the current syntax node to a given node type from the different base node protocol hierarchy than ``DeclSyntaxProtocol``.
+  /// Force-casts the current syntax node to a given node type from a base node protocol hierarchy other than
+  /// ``DeclSyntaxProtocol``.
   ///
   /// - Returns: This method will always trigger a runtime crash and never return.
   ///
```

Comment:
This comment is no longer correct

Reply:
Thanks. I reverted my old PR and apparently didn't copy across all the comment changes 😅